workers/unity/Packages/io.improbable.gdk.buildsystem/.codegen/Source/WorkerGenerationJob.cs

```diff
@@ -20,58 +20,85 @@ public class WorkerGenerationJob : CodegenJob
+            logger.Info($"Generating {workerFileName}");
             var unityWorkerMenuGenerator = new UnityWorkerMenuGenerator();
             var workerCode = unityWorkerMenuGenerator.Generate(workerTypesToGenerate);
-            Content.Add(Path.Combine(relativeEditorPath, workerFileName), workerCode);
+            AddContent(Path.Combine(relativeEditorPath, workerFileName), workerCode);
 
+            logger.Info($"Generating {buildSystemFileName}");
             var buildSystemAssemblyGenerator = new BuildSystemAssemblyGenerator();
             var assemblyCode = buildSystemAssemblyGenerator.Generate();
-            Content.Add(Path.Combine(relativeOutputPath, buildSystemFileName), assemblyCode);
+            AddContent(Path.Combine(relativeOutputPath, buildSystemFileName), assemblyCode);
 
-            Content.Add(Path.Combine(relativeEditorPath, workerListFileName), string.Join(Environment.NewLine, workerTypesToGenerate));
+            logger.Info($"Generating {workerListFileName}");
+            AddContent(Path.Combine(relativeEditorPath, workerListFileName), string.Join(Environment.NewLine, workerTypesToGenerate));
         }
 
         private List<string> ExtractWorkerTypes(string path)
         {
+            logger.Info($"Extracting worker types from {path}");
+
             var workerTypes = new List<string>();
-            var fileNames = Directory.EnumerateFiles(path, "*.json");
+
+            var fileNames = Directory.EnumerateFiles(path, "*.json").ToList();
+            logger.Trace($"Found {fileNames.Count()} worker json files: {string.Join(", ", fileNames)}");
+
             foreach (var fileName in fileNames)
             {
-                string text = File.ReadAllText(fileName);
+                logger.Trace($"Extracting worker type from {fileName}");
+                var text = File.ReadAllText(fileName);
                 if (!text.Contains(workerTypeFlag))
                 {
-                    Console.WriteLine($"{fileName} does not contain the following flag: {workerTypeFlag}");
+                    logger.Warn($"{fileName} does not contain the '{workerTypeFlag}' flag");
                     continue;
                 }
 
                 var jsonRep = JObject.Parse(text);
                 var arguments = jsonRep.SelectToken("external.default.windows.arguments");
                 if (arguments == null)
                 {
-                    Console.WriteLine($"Could not navigate to external > default > windows > arguments in {fileName}");
+                    logger.Warn($"Could not navigate to external > default > windows > arguments in {fileName}");
                     continue;
                 }
 
                 for (var i = 0; i < arguments.Count(); i++)
                 {
                     if (workerTypeFlag.Equals(arguments[i].ToString()))
                     {
-                        workerTypes.Add(arguments[i + 1].ToString());
+                        var workerType = arguments[i + 1].ToString();
```

Comment:
Wow this indexing operation is unsafe lol

Reply:
lol yeah - martijn made the same point

but because the args are always in pairs of key/value we've never encountered the error

(rest assured I've fixed it locally)